src/cmd/build.js

```diff
@@ -103,15 +111,49 @@ async function defaultPackageCreator({
   let packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   let extensionPath = path.join(artifactsDir, packageName);
-  let stream = createWriteStream(extensionPath);
 
-  stream.write(buffer, () => stream.end());
+  let destinationFileExists = await fs.stat(extensionPath)
+    .then((stat)=>{
+      return stat.isFile();
+    })
+    .catch(()=>{
```

Comment:
I think that plan sounds good but you need to use [isErrorWithCode()](https://github.com/mozilla/web-ext/blob/b050f119ebda01c89c33fd8572e6fd6d40ee7adc/src/util/artifacts.js#L19) to make sure the error code means 'file path already exists' etc.